docs/plugins/development.md

```diff
@@ -305,3 +306,40 @@ A `PluginNavMenuButton` has the following attributes:
 * `color` - Button color (one of the choices provided by `ButtonColorChoices`)
 * `icon_class` - Button icon CSS class
 * `permission` - The name of the permission required to display this button (optional)
+
+## Template Content
+
+Plugins can inject custom content into certain areas of the detail views of applicable models. This is accomplished by subclassing `PluginTemplateContent`, designating a particular NetBox model, and defining the desired methods to render custom content. Four methods are available:
+
+* `left_page()` - Inject content on the left side of the page
+* `right_page()` - Inject content on the right side of the page
+* `full_width_page()` - Inject content across the entire bottom of the page
+* `buttons()` - Add buttons to the top of the page
+
+Each of these methods must return HTML content suitable for inclusion in the object template. Two instance attributes are available for context:
+
+* `self.obj` - The object being viewed
+* `self.context` - The current template context
```

Comment:
While this is true, it might be worth mentioning this "original" context is available in the `render()` method as `obj_context`

Reply:
Why are we passing the original context as a single variable, as opposed to just passing it directly (after adding the `obj` variable)?

Reply:
Looking at this further, we should restrict the template context to a prescribed set of variables, since we make no guarantees that any of the context data present in the current release will always be available. I think it makes sense to limit the context available to a plugin to:

* The object being viewed
* The current request
* The plugin's configuration